### DISCUSSION TITLE
HEIC sequence compression on MCAP data

### DISCUSSION THREAD

**Discussion by russkel** [foxglove/mcap#1074] [General] [answered]
Hello,

We are thinking about using MCAP for storing training data for various projects. This has got me thinking about compression using newer formats. The HEIC file format supports sequences of images and presumably improved compression ratios given the redundancy between frames. Could this be implemented at an MCAP level using the compression API or does the compression only do chunks of data vs individual messages?

**Comment by achim-k** [comment, marked as answer]
MCAP compresses chunks of of data but not individual messages. You will have to compress messages before writing them to the mcap file. This is for instance done in the demo application on https://mcap.dev (when scrolling to the bottom).

![image](https://github.com/foxglove/community/assets/9250155/43ce3cee-83c8-4a65-b7a2-b1b657017757)

Images can be recorded with either h264 or jpeg compression with the former yielding smaller file sizes as it uses delta frames.

```
channels:
	(0) mouse        89 msgs (38.59 Hz)   : MouseEvent [jsonschema]              
	(1) camera_h264  70 msgs (30.36 Hz)   : foxglove.CompressedVideo [protobuf]  
	(2) camera_jpeg  69 msgs (29.92 Hz)   : foxglove.CompressedImage [protobuf]  
```

**Reply by russkel** [reply]
Okay so I will ignore mcap compression functionality. I will have a play with image sequences and storing individual images in mcap!

Thanks.
